server/cli/src/cli.rs

```diff
@@ -440,7 +440,9 @@ async fn main() -> anyhow::Result<()> {
                     let code = manifest.code;
 
                     let version = manifest.version;
-                    let id = format!("{code}_{version}");
+                    let id_version = str::replace(&version, ".", "_");
```

Comment:
I'm wondering if we are better to just use a UUID here? Consistent with all our other processes?
I quite like this approach of having a readable ID, but wondering if it creates other problems...

Reply:
What other problems do you imagine James? I quite liked the idea of the version being included, if ever needing to troubleshoot (It helped me realise why I had duplicates from v2.2 😁)

Reply:
I think this is the blog post I'm looking for?
https://blog.ploeh.dk/2024/06/03/youll-regret-using-natural-keys/

I'm on the fence with this one, as like you I kind of line having it how it is. 

I just know that trying to produce reliable ids like this can have unintended consequences (such as we've just seen), and the assumptions behind them might not always hold in the future?

I'm fine with it how it is personally, just need the migration to avoid the issues for tests in the RC build.

Reply:
I am ok with underscore for this fix (would also include is_custom in 2.4)

I think with pre generated ids we might have issue with uniqueness (especially when there are custom reports in separate repos)